### PR TITLE
Prometheus on tendrl2-server

### PR DESCRIPTION
**Pull request by Daniel-Pivonka**
- Need to add a new vm in the above setup
- This new VM should be of the same image as the other vm's currently in tendrl2-vagrant
- On this new VM, tendrl2-vagrant should be able to install, configure and start https://prometheus.io/

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -159,7 +159,7 @@ Vagrant.configure(2) do |config|
       # Provider-independent options
       machine.vm.hostname = "gd2-#{node_index}"
       machine.vm.synced_folder '.', '/vagrant', disabled: true
-      machine.vm.synced_folder '/home/smukherjee/dev/redhat/tendrl/golang/src/github.com/gluster/glusterd2/build/', '/vagrant'
+      machine.vm.synced_folder '/home/dpivonka/golang/src/github.com/gluster/glusterd2/build/', '/vagrant'
```
Comment (r0h4n) [inline Vagrantfile]: the path to find glusterd2 builds on vagrant vms, could you make this configurable?

@shirshendu comments?

File: Vagrantfile [hunk 2]
```diff
@@ -265,4 +265,54 @@ Vagrant.configure(2) do |config|
       end
     end
   end
+
+  config.vm.define "prometheus" do | prometheus |
+    prometheus.vm.hostname = "prometheus"
```
Comment (r0h4n) [inline Vagrantfile]: lets call this vm the "tendrl server"

File: Vagrantfile (hunk 1 above)
Comment (Daniel-Pivonka) [reply Vagrantfile]: that shouldn't be hard. I can make it so its stored in the tendrl.conf.yml

File: Vagrantfile (hunk 2 above)
Comment (Daniel-Pivonka) [reply Vagrantfile]: alright

**Comment by shirshendu** [comment]
Due to some rapid iterations I needed for the PoC, I ended up duplicating some of the work in this PR. Could we add a single PR just for the prometheus deployment, along the lines of how services are deployed in the current master?

**Comment by r0h4n** [comment]
@Daniel-Pivonka please edit this PR to only contain the prometheus installation.

**Comment by Daniel-Pivonka** [comment]
i pulled the changes made upstream into my fork and moved everything around and tried to follow the nameing conventions. ive added the Prometheus stuff i made into that.

**Comment by Daniel-Pivonka** [comment]
the tendrl2.server role has code that is missing some variable so the playbook does not complete(etcd vars).
also the ansible.cfg gave me permission issues even after changing the log location to a file on my pc.

File: Vagrantfile [hunk 3]
```diff
@@ -240,9 +240,9 @@ Vagrant.configure(2) do |config|
         machine.vm.provision :deploy_tendrl, type: :ansible do |ansible|
           ansible.limit = 'all'
           ansible.groups = {
-            'tendrl2-server' => ['tendrl2-server']
+            'tendrl2-servers' => ['tendrl2-server']
           }
-          ansible.playbook = 'ansible/tendrl2-server.yml'
+          ansible.playbook = 'ansible/prepare-tendrl2-server.yml'
```
Comment (shirshendu) [inline Vagrantfile]: We already have a playbook at ansible/tendrl2-server.yml. Why do we want to make a new one?

**Pull request merged**